eng/scripts/InstallVisualStudio.ps1

```diff
@@ -54,35 +48,57 @@ mkdir $intermedateDir -ErrorAction Ignore | Out-Null
 
 $bootstrapper = "$intermedateDir\vsinstaller.exe"
 $ProgressPreference = 'SilentlyContinue' # Workaround PowerShell/PowerShell#2138
-Invoke-WebRequest -Uri "https://aka.ms/vs/16/pre/vs_$($Edition.ToLowerInvariant()).exe" -OutFile $bootstrapper
+Invoke-WebRequest -Uri "https://aka.ms/vs/16/release/vs_$($Edition.ToLowerInvariant()).exe" -OutFile $bootstrapper
+
+$productId = "Microsoft.VisualStudio.Product.$Edition"
+if (-not $InstallPath) {
+    $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (Test-Path $vsWhere)
+    {
+        $InstallPath = &$vsWhere -version '[16,17)' -latest -prerelease -products $productId -property installationPath
+    }
+}
 
 if (-not $InstallPath) {
-    # $InstallPath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\$Edition"
-    $InstallPath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Preview"
+    $InstallPath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\$Edition"
 }
 
 # no backslashes - this breaks the installer
 $InstallPath = $InstallPath.TrimEnd('\')
 
 [string[]] $arguments = @()
-
 if (Test-path $InstallPath) {
     $arguments += 'modify'
 }
 
+$responseFile = "$PSScriptRoot\vs.json"
+if ("$Edition" -eq "BuildTools") {
+    $responseFile = "$PSScriptRoot\vs.buildtools.json"
+}
+
 $arguments += `
-    '--productId', "Microsoft.VisualStudio.Product.$Edition", `
+    '--productId', $productId, `
     '--installPath', "`"$InstallPath`"", `
-    '--in', "$PSScriptRoot\vs.json", `
+    '--in', "`"$responseFile`"", `
     '--norestart'
 
 if ($Passive) {
     $arguments += '--passive'
 }
+if ($Quiet) {
+    $arguments += '--quiet', '--wait'
+}
 
 Write-Host ""
 Write-Host "Installing Visual Studio 2019 $Edition" -f Magenta
 Write-Host ""
 Write-Host "Running '$bootstrapper $arguments'"
 
-& $bootstrapper @arguments
+$process = Start-Process -FilePath "$bootstrapper" -ArgumentList $arguments `
+    -PassThru -RedirectStandardError "$intermedateDir\errors.txt" -Verbose -Wait
```

Comment:
Interesting, so if you don't redirect it to a file to then write it out it doesn't make it to the console output? Also, if you're taking this approach probably smart to remove the errors.txt before running this in case someone cancelled the build.

Reply:
The file is removed just before exiting this script.